src/main/java/com/hubspot/jinjava/lib/filter/JoinFilter.java

```diff
@@ -52,17 +56,36 @@ public Object filter(Object var, JinjavaInterpreter interpreter, String... args)
     }
 
     ForLoop loop = ObjectIterator.getLoop(var);
+    boolean first = true;
     while (loop.hasNext()) {
       Object val = loop.next();
 
       if (attr != null) {
         val = interpreter.resolveProperty(val, attr);
       }
 
-      vals.add(Objects.toString(val, ""));
+      try {
+        if (!first) {
+          stringBuilder.append(separator);
+        } else {
+          first = false;
+        }
+        stringBuilder.append(Objects.toString(val, ""));
+      } catch (OutputTooBigException ex) {
+        interpreter.addError(new TemplateError(ErrorType.WARNING,
```

Comment:
This seems FATAL.

Reply:
Not necessarily FATAL IMO since it will still output a string, just shorter than expected.